python/cudf/cudf/core/column_accessor.py

```diff
@@ -523,14 +523,19 @@ def rename_column(x):
                 raise IndexError(
                     f"Too many levels: Index has only 1 level, not {level+1}"
                 )
+
             if isinstance(mapper, Mapping):
-                new_names = (
+                new_col_names = [
                     mapper.get(col_name, col_name) for col_name in self.keys()
-                )
+                ]
```

Comment:
It's not immediately obvious to me - why opt for a list instead of a tuple here?

Reply:
This doesn't change a tuple to a list, it changes a _generator_ to a list. There's no equivalent to a list comprehension for a tuple because the syntax `(x for x in ...)` doesn't create a tuple, it creates a generator. The reason this change is necessary is that you cannot query the length of a generator, so the error check below won't work.

Aside: If you're not familiar with generators, they are what you get when you write a function that uses `yield` instead of `return`. The purpose of generators is not to materialize the entire set of data at once, but to instead produce the data one at a time. One of the most obvious uses of this is to save memory, but there are also lots of other reasons you might prefer a generator to a list such as wanting to produce an arbitrary number of elements from an infinite sequence. Here's a simple example illustrating the difference:
```
>>> def f(x):                                                                                                                                                                                     
...     print(f"Called f({x})")                                                                       
...     return x**2
>>> for _ in (f(x) for x in range(3)):
...     print("Hello")                                                                                
... 
Called f(0)
Hello
Called f(1)
Hello
Called f(2)
Hello
>>> for _ in [f(x) for x in range(3)]:                                                                
...     print("Hello")
... 
Called f(0)
Called f(1)
Called f(2)
Hello
Hello
Hello
```